pkg/secretless/config/v2/v1_service.go

```diff
@@ -71,35 +78,60 @@ func newV1Service(v2Svc Service) (ret *v1Service, err error) {
 
 	// Apply protocol specific config
 
-	if err = ret.applyProtocolConfig(v2Svc.ProtocolConfig); err != nil {
+	if err = ret.applyProtocolConfig(v2Svc); err != nil {
 		return nil, err
 	}
 
 	return ret, nil
 }
 
-func (v1Svc *v1Service) applyProtocolConfig(cfgBytes []byte) error {
+func (v1Svc *v1Service) applyProtocolConfig(v2Svc Service) error {
+	cfgBytes := v2Svc.ConnectorConfig
 	switch v1Svc.Listener.Protocol {
 	case "http":
-		if err := v1Svc.configureHTTP(cfgBytes); err != nil {
-			return err
+		if len(cfgBytes) == 0 {
+			return fmt.Errorf("empty http config")
 		}
-	}
-	return nil
-}
 
-func (v1Svc *v1Service) configureHTTP(cfgBytes []byte) error {
-	if len(cfgBytes) == 0 {
-		return fmt.Errorf("empty http config")
-	}
+		var httpHandlerType string
+		if IsHTTPConnector(v2Svc.Connector) {
+			httpHandlerType = v2Svc.Connector
+		} else {
+			// This temp struct makes it possible to parse 'authenticateURLsMatching' as
+			// string or []string
+			tempCfg := &struct {
+				AuthenticationStrategy string `yaml:"authenticationStrategy"`
+			}{}
+			err := yaml.Unmarshal(v2Svc.ConnectorConfig, tempCfg)
+			if err != nil {
+				return fmt.Errorf("failed to parse 'config' key for service '%s': %s", v2Svc.Name, err)
+			}
+
+			err = validation.ValidateStruct(
+				tempCfg,
+				validation.Field(
+					&tempCfg.AuthenticationStrategy,
+					validation.Required,
+					validation.In(HTTPAuthenticationStrategies...),
+				),
+			)
+
+			if err != nil {
+				return fmt.Errorf("failed validation on 'config' key for service '%s': %s", v2Svc.Name, err)
+			}
+
+			httpHandlerType = tempCfg.AuthenticationStrategy
+		}
 
-	httpCfg, err := newHTTPConfig(cfgBytes)
-	if err != nil {
-		return err
-	}
+		httpCfg, err := newHTTPConfig(cfgBytes)
+		if err != nil {
+			return err
+		}
 
-	v1Svc.Handler.Match = httpCfg.AuthenticateURLsMatching
-	v1Svc.Handler.Type = httpCfg.AuthenticationStrategy
+		v1Svc.Handler.Match = httpCfg.AuthenticateURLsMatching
+		v1Svc.Handler.Type = httpHandlerType
 
+		return nil
+	}
```

Comment:
Not a blocker but this method is a bit gnarly though I don't have any specific guidance on how to improve it :/

Reply:
One small thing: I've read now in multiple Go best practice guides that you can (almost always) rewrite logic to remove `else` clauses from your logic, and I agree that this improves readability.